connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -150,10 +156,31 @@ private void loadInitialConsumerGroups()
 
     List<String> findConsumerGroups()
             throws InterruptedException, ExecutionException {
-        return listConsumerGroups().stream()
+        List<String> filteredGroups = listConsumerGroups().stream()
                 .map(ConsumerGroupListing::groupId)
-                .filter(this::shouldReplicate)
+                .filter(this::shouldReplicateByGroupFilter)
                 .collect(Collectors.toList());
+
+        List<String> checkpointGroups = new LinkedList<>();
+        List<String> irrelevantGroups = new LinkedList<>();
+
+        for (String group : filteredGroups) {
+            Set<String> consumedTopics = listConsumerGroupOffsets(group).keySet().stream()
+                    .map(TopicPartition::topic)
+                    .filter(this::shouldReplicateByTopicFilter)
```

Comment:
Is this backward compatible? Can this break use-cases where user is using MM2 to sync the offsets but not necessarily topics?

Reply:
I don't believe MM2 ever supported that. From [KIP-382](https://cwiki.apache.org/confluence/display/KAFKA/KIP-382%3A+MirrorMaker+2.0#KIP382:MirrorMaker2.0-InternalTopics), regarding checkpoints (emphasis mine):

> The connector will periodically query the source cluster for all committed offsets from all consumer groups, **filter for those topics being replicated**, and emit a message to a topic

IIUC, MM2 also performs offset syncing based on the contents of the offset syncs topic, which is only populated by the source connector (i.e., the connector that replicates topics).

@blacktooth have you experienced otherwise?

Reply:
@blacktooth I'll mark this conversation resolved so that users don't get the wrong idea about MM2; feel free to unresolve if you believe there's still an issue.